lambda-runtime/src/main/scala/feral/lambda/runtime/LambdaRuntime.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda
+package runtime
+
+import cats.{ApplicativeError, MonadThrow}
+import cats.data.OptionT
+import cats.effect.Temporal
+import cats.syntax.all._
+import cats.effect.syntax.all._
+import cats.effect.kernel.{Async, Concurrent, Sync}
+import io.circe.Json
+import org.http4s.Method.POST
+import org.http4s.client.Client
+import org.http4s.circe._
+
+import scala.concurrent.duration.FiniteDuration
+import java.util.concurrent.TimeUnit
+import org.http4s.{EntityEncoder, Request, Uri}
+import org.http4s.client.dsl.Http4sClientDsl
+import io.circe._
+import cats.effect.kernel.Outcome._
+import cats.effect.std.Env
+import io.circe.syntax.EncoderOps
+import org.http4s.implicits.http4sLiteralsSyntax
+
+import scala.concurrent.CancellationException
+
+object FeralLambdaRuntime {
+
+  final val ApiVersion = "2018-06-01"
+
+  // TODO refactor into setup and per-request phases, need to find boundary between two phases
+  def apply[F[_]](client: Client[F])(handler: (Json, Context[F]) => F[Json])(implicit F: Temporal[F], env: LambdaRuntimeEnv[F]): F[Unit] = {
```

Comment:
Btw, thanks for raising the issue about loading the handler function. After thinking about it, you are right that we need a small adjustment here actually.

```suggestion
  def apply[F[_]](client: Client[F])(handler: Resource[F, (Json, Context[F]) => F[Json]])(implicit F: Temporal[F], env: LambdaRuntimeEnv[F]): F[Unit] = {
```

Instead of requesting the handler, we should request a `Resource` that is used to acquired the handler. This way, if there are any errors during that process, we can log them correctly. It also leaves the door open for someone to implement dynamic loading, but we shouldn't worry about that here.

Reply:
I believe the JVM runtime uses reflection based loading of the handler class and the JS runtime uses dynamic loading. Errors can occur during these steps so that's where the initialization error URL is used.